src/quo2/components/list_items/channel.cljs

```diff
@@ -1,26 +1,72 @@
+            [status-im.utils.handlers :refer [>evt]]
+            [status-im.utils.utils :as utils]))
+
+(def eth-token-img (js/require "../resources/images/tokens/mainnet/ETH.png"))
+(def dai-token-img (js/require "../resources/images/tokens/mainnet/DAI.png"))
+(def knc-token-img (js/require "../resources/images/tokens/mainnet/KNC.png"))
+(def mana-token-img (js/require "../resources/images/tokens/mainnet/MANA.png"))
+(def rare-token-img (js/require "../resources/images/tokens/mainnet/RARE.png"))
 
 (defn list-item [{:keys [name locked? mentions-count unread-messages?
-                         muted? is-active-channel? emoji channel-color]
+                         muted? is-active-channel? emoji channel-color gated]
                   :or {channel-color colors/primary-50}}]
-  [rn/view {:style (merge {:height          48
-                           :display         :flex
-                           :border-radius   12
-                           :flex-direction  :row
-                           :justify-content :space-between
-                           :align-items     :center
-                           :width           "100%"
-                           :padding-left    12
-                           :padding-right   12}
-                          (when is-active-channel?
-                            {:background-color (colors/theme-alpha channel-color 0.05 0.05)}))}
+  [rn/touchable-opacity {:style (merge {:height          48
+                                        :display         :flex
+                                        :border-radius   12
+                                        :flex-direction  :row
+                                        :justify-content :space-between
+                                        :align-items     :center
+                                        :width           "100%"
+                                        :padding-left    12
+                                        :padding-right   12}
+                                       (when is-active-channel?
+                                         {:background-color (colors/theme-alpha channel-color 0.05 0.05)}))
+                         :on-press (when locked?
+                                     #(>evt
+                                       [:bottom-sheet/show-sheet
+                                        {:content
+                                         (constantly [token-gating/token-gating
+                                                      {:channel {:name name
+                                                                 :community-color (colors/custom-color :pink 50)
+                                                                 :emoji emoji
+                                                                 :emoji-background-color channel-color
+                                                                 :on-enter-channel (fn [] (utils/show-popup "Entered channel" "Wuhuu!! You successfully entered the channel :)"))
```

Comment:
1. > Entered channel" "Wuhuu!! You successfully entered the channel :)

testing?
2. Maybe create function for this dispatch and assign it to on-press (more readable)